packages/typespec-ts/package.json

```diff
@@ -58,19 +58,20 @@
     "pacote": "15.1.3",
     "@types/pacote": "11.1.5"
   },
+  "peerDependencies": {
+    "@azure-tools/typespec-azure-core": ">=0.30.0 <1.0.0",
+    "@azure-tools/typespec-client-generator-core": ">=0.30.0 <1.0.0",
+    "@typespec/compiler": ">=0.44.0 <1.0.0",
+    "@typespec/http": ">=0.44.0 <1.0.0",
+    "@typespec/rest": ">=0.44.0 <1.0.0",
+    "@typespec/versioning": ">=0.44.0 <1.0.0"
```

Comment:
why these libraries are not necessarily required for typescript emiter ? https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependencies  @joheredi Could you help confirm if this is ok with you ?

Reply:
We normally recommend against peer deps for most packages, due to the complexity and confusion surrounding how to satisfy them. I'm not sure if TypeSpec decided differently for the purposes of emitters.

@bterlson can you help offer guidance?